### PR TITLE
Check whether standby is streaming from master

### PR DESCRIPTION
**Pull request by UnderGreen**

### CODE REVIEW

File: cmd/keeper/keeper.go [hunk 1]
```diff
@@ -1039,18 +1039,35 @@ func (p *PostgresKeeper) resyncIfNotReady(followed *cluster.KeeperState, initial
...
+		if err = p.resyncAndStart(followed, initialized, started); err != nil {
+			return trace.Wrap(err)
 		}
-		if err = p.pgm.Start(); err != nil {
-			return trace.Wrap(err, "error starting PostgreSQL instance")
+	}
+
+	if err = p.pgm.IsStreaming(); err != nil {
+		log.Info("Standby is not accepting connections. It's probably waiting for unavailable WALs. Forcing a full resync.")
+		log.Debugf("Error: %v", err)
```
Comment (a-palchikov) [inline cmd/keeper/keeper.go]: Is this going to spam logs if you logged this as `Warn`? How interesting would be this error for possible troubleshooting?

File: pkg/postgresql/postgresql.go [hunk 2]
```diff
@@ -43,6 +43,8 @@ const (
 	sleepBetweenChecks = 200 * time.Millisecond
 )
 
+var log = capnslog.NewPackageLogger("github.com/gravitational/pkg/postgresql", "postgresql")
```
Comment (a-palchikov) [inline pkg/postgresql/postgresql.go:46]: The package name would be `github.com/gravitational/stolon/pkg/postgresql`

File: pkg/postgresql/utils.go [hunk 3]
```diff
@@ -362,3 +363,25 @@ func GetTimelinesHistory(ctx context.Context, timeline uint64, replConnParams Co
 func IsValidReplSlotName(name string) bool {
 	return ValidReplSlotName.MatchString(name)
 }
+
+// isStreaming returns error if the PostgreSQL is not streaming WALs from master
```
Comment (a-palchikov) [inline pkg/postgresql/utils.go]: this should be phrased in a positive way first - i.e. what the purpose is and then describe the error conditions (if any):
```go
// isStreaming determines whether the PostgreSQL is streaming WALs from (the,a?) master.
// Returns NotFound if no receiver configured for WALs.
```
Please correct the comment if it does not reflect the full semantics.

File: cmd/keeper/keeper.go [hunk 4]
```diff
@@ -1039,18 +1039,35 @@ func (p *PostgresKeeper) resyncIfNotReady(followed *cluster.KeeperState, initial
...
-			return trace.Wrap(err, "failed to full resync from followed instance")
+		log.Info("Standby is not accepting connections. Forcing a full resync.")
+		if err = p.resyncAndStart(followed, initialized, started); err != nil {
+			return trace.Wrap(err)
 		}
-		if err = p.pgm.Start(); err != nil {
-			return trace.Wrap(err, "error starting PostgreSQL instance")
+	}
+
+	if err = p.pgm.IsStreaming(); err != nil {
```
Comment (a-palchikov) [inline cmd/keeper/keeper.go]: Seems like this is going to `resyncAndStart` if either `!ready` or `!IsStreaming` - did you mean to check them and restart sequentially or can the `resyncAndStart` be done once, i.e.:
```go
if !ready || p.pgm.IsStreaming() != nil { // Note - not actual code, but an example to demonstrate the point
	p.resyncAndStart(...)
}
```

**Review by a-palchikov — APPROVED** [review]

File: cmd/keeper/keeper.go (hunk 1 above)
Comment (UnderGreen) [reply cmd/keeper/keeper.go]: This line supposed to be deleted. Left it after debugging the code.

**Pull request merged**